hunts/src/use-global-filter-fixed.js

```diff
@@ -0,0 +1,219 @@
+import React from 'react'
+
+import {
+  actions,
+  useMountedLayoutEffect,
+  functionalUpdate,
+  useGetLatest,
+} from 'react-table'
+
+const filterTypes = {};
+
+// Actions
+actions.resetGlobalFilter = 'resetGlobalFilter'
+actions.setGlobalFilter = 'setGlobalFilter'
+
+export const useGlobalFilter = hooks => {
+  hooks.stateReducers.push(reducer)
+  hooks.useInstance.push(useInstance)
+}
+
+useGlobalFilter.pluginName = 'useGlobalFilter'
+
+// Copied from utils.js in the original project
+function isFunction(a) {
+  if (typeof a === 'function') {
+    return a
+  }
+}
+
+function getFirstDefined(...args) {
+  for (let i = 0; i < args.length; i += 1) {
+    if (typeof args[i] !== 'undefined') {
+      return args[i];
+    }
+  }
+}
+
+function getFilterMethod(filter, userFilterTypes, filterTypes) {
+  return (
+    isFunction(filter) ||
+    userFilterTypes[filter] ||
+    filterTypes[filter] ||
+    filterTypes.text
+  );
+}
+
+function shouldAutoRemoveFilter(autoRemove, value, column) {
+  return autoRemove ? autoRemove(value, column) : typeof value === 'undefined'
+}
+
+// End of utils.js copying
+
+function reducer(state, action, previousState, instance) {
+  if (action.type === actions.resetGlobalFilter) {
```

Comment:
would recommend using a switch statement on action.type

Reply:
To be clear I didn't write this code and don't intend on modifying it. This file is modified from the react-table library, you can see the changes in the commit log.